### PR TITLE
fix `invul_mode` affect beta light amp

### LINKED ISSUE #1583: Gray Invulnerability effect overrides Beta's Light-Amp Visor effect
_Tested in the [latest master artifact (x64)](https://github.com/fabiangreffrath/woof/actions/runs/8242570467)._

As the title says; when playing with Beta Emulation, the Light Amp Visor makes use of a different colormap, namely index 32. Said index is the one that's changed in-game by the _Invulnerability Effect_ setting (after all, index 32 is used for Invulnerability in the final game), and as a result, it overrides the Visor effect.

It looks like Invulnerability has no screen effect during Beta Emulation, so I believe that the fix would be to not alter the colormap during it.

I must admit that I'm surprised by having found this bug. What are the chances?

### PR DESCRIPTION
**Pull request by rfomin**
Fix #1583

### CODE REVIEW

File: src/r_data.c [hunk 1]
```diff
@@ -852,7 +852,7 @@ static byte invul_orig[256];
 
 void R_InvulMode(void)
 {
-  if (colormaps == NULL)
+  if (colormaps == NULL || beta_emulation)
```
Comment (fabiangreffrath) [inline src/r_data.c:855]: Should the menu switch get disabled as well?

Comment (rfomin) [reply src/r_data.c:855]: We could introduce the `S_BETA` flag and disable a lot of things that don't work in beta mode, but I don't think it's worth it. What if we remove beta emulation altogether?

Comment (fabiangreffrath) [reply src/r_data.c:855]: > a lot of things that don't work in beta mode

What would that be?

Comment (rfomin) [reply src/r_data.c:855]: > > a lot of things that don't work in beta mode
> 
> What would that be?

Complevels, "Preferences" tab in the "Weapons" menu, screen flashes. Probably more, I can't remember.

Comment (fabiangreffrath) [reply src/r_data.c:855]: Direct aiming for missiles.

Comment (rfomin) [reply src/r_data.c:855]: So should we disable the menu items for beta mode? I don't think it helps much. Currently, if users want to experience beta mode, they have to do the following:
1. Download mbf.zip for BETAGRPH.WAD and BETALEVL.WAD
2. Run the game with this command line: `woof -beta -complevel mbf -iwad doom.wad -file BETALEVL.WAD`

At this point it is easier (and more fun) to run press beta in DosBox.

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**